src/inertia.js

```diff
@@ -10,7 +10,9 @@ export default {
 
   init(page, setPage) {
     this.version = page.version
-    this.setPage = setPage
+    this.setPage = (page, token = this.createToken()) => {
+      return setPage(page, () => token === this.token)
+    }
```

Comment:
I wonder, does it really make sense to even call `setPage()` if we know we don't want to update? I wonder if this would be better:

```js
this.setPage = (page, token = this.createToken()) => {
  if (token === this.token) {
    return setPage(page)
  }
}
```

This is totally untested, but it would remove the need to have this handled in the specific adapters.